lib/ASN1/Universal/OctetString.php

```diff
@@ -47,18 +47,27 @@ public function getType()
 
     protected function calculateContentLength()
     {
+        if (is_null($this->value)) {
+            return 0;
+        }
         return strlen($this->value) / 2;
     }
 
     protected function getEncodedValue()
     {
         $value = $this->value;
+        if (is_null($value)) {
+            return '';
+        }
+        if (strlen($value) & 1) {
+            $value = '0' . $value;
+        }
         $result = '';
 
         //Actual content
         while (strlen($value) >= 2) {
             // get the hex value byte by byte from the string and and add it to binary result
-            $result .= chr(hexdec(substr($value, 0, 2)));
+            $result .= @chr(hexdec(substr($value, 0, 2)));
```

Comment:
We may want to replace `chr($val)` with `pack('C', $val)`.

https://paragonie.com/blog/2017/02/cryptographically-secure-php-development#chr